src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PipelineRuntimeMacros.java

```diff
@@ -0,0 +1,295 @@
+    try {
+      WebElement rowEl = Helper.locateElementByCssSelector(
+        Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+        Helper.getCssSelectorByDataTestId(index) + " " +
+        Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_KEY_SELECTOR));
+      Assert.assertFalse(ElementHelper.isElementDisplayed(rowEl));
+    } catch (NoSuchElementException e) {
+      // Do nothing; this is the expected outcome
+    }
+  }
+
+  @When("Add a preview runtime argument from row {string}")
+  public void addARuntimeArgumentFromRowInt(String index) {
+    ElementHelper.clickOnElement(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId("add-row")
+    ));
+  }
+
+  @When("Enter preview runtime argument key {string} in row {string}")
+  public void enterRuntimeArgumentKeyStringInRowInt(String key, String index) {
+    ElementHelper.sendKeys(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_KEY_SELECTOR) + " input"), key);
+  }
+
+  @When("Delete preview runtime argument row {string}")
+  public void deleteRuntimeArgumentRowInt(String index) {
+    ElementHelper.clickOnElement(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId("remove-row")
+    ));
+  }
+
+  @When("Enter preview runtime argument value {string} in row {string}")
+  public void enterRuntimeArgumentValueStringInRowInt(String value, String index) {
+    ElementHelper.sendKeys(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_PREVIEW_SELECTOR) + " " +
+      Helper.getCssSelectorByDataTestId(index) + " " +
+      Helper.getCssSelectorByDataTestId(Constants.RUNTIME_ARGS_VALUE_SELECTOR) + " textarea"), value);
+  }
+
+  @When("Click run button from preview runtime arguments")
+  public void clickRunButtonFromRuntimeArguments() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("preview-configure-run-btn"));
+  }
+
+  @Then("Pipeline banner is shown with message {string}")
+  public void pipelineBannerIsShownWithMessage(String message) {
+    WaitHelper.waitForElementToBePresent(By.cssSelector(Helper.getCssSelectorByDataTestId("valium-banner-hydrator")));
+    Assert.assertTrue(
+      ElementHelper.getElementText(Helper.locateElementByTestId("valium-banner-hydrator")).contains(message));
+  }
+
+  @Given("Pipeline deployed from file {string}")
+  public void pipelineDeployedFromFileString(String filename) {
```

Comment:
is the reason you don't want to use `deployAndTestPipeline` avoiding providing the pipelineName?

Reply:
No, I'll switch to that

Reply:
Actually, `deployAndTestPipeline` isn't working for any test locally